integration-tests/oidc-mock/src/test/java/io/quarkus/it/keycloak/KeycloakTestResource.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkus.it.keycloak;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.ws.rs.core.MediaType;
+
+import org.jboss.logging.Logger;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+public class KeycloakTestResource implements QuarkusTestResourceLifecycleManager {
+
+    private static final Logger LOG = Logger.getLogger(KeycloakTestResource.class);
+
+    private WireMockServer server;
+
+    @Override
+    public Map<String, String> start() {
+
+        server = new WireMockServer();
+        server.start();
+
+        WireMock.stubFor(
+                get(urlEqualTo("/auth/realms/quarkus/.well-known/openid-configuration"))
+                        .willReturn(aResponse()
+                                .withHeader("Content-Type", MediaType.APPLICATION_JSON)
+                                .withBody("{\n" +
+                                        "    \"introspection_path\": \"" + server.baseUrl()
+                                        + "/auth/realms/quarkus/protocol/openid-connect/token/introspect\"\n" +
+                                        "}")));
+
+        // define the mock for the introspect endpoint
+        WireMock.stubFor(WireMock.post("/auth/realms/quarkus/protocol/openid-connect/token/introspect").willReturn(WireMock
```

Comment:
@cemnura what should be done here is to check if the [incoming token](https://tools.ietf.org/html/rfc7662#section-2.1) is equal to `123` and if yes then return this response, it should be done anyway, without the expiried token check, as it has to be tested that it is indeed a `123` token which is being introspected.
Once it works then it will be easy to check if this token is equal to `456` and if yes then return the response with the expired value set to some low value, less that the issued at time etc :-)

Reply:
I think I got it. 

When there is a Authorization header with a value equal to 123 is sent then the current valid response is returned. In other cases then the response returns the response with the expired value set to some low value.

Reply:
Hi @cemnura it will be in a POST body: `token=123` etc